File: UI/material-kit-react-main/material-kit-react-main/src/types/property/facility.ts
import { RoleName } from '../auth/userRole';

export enum FacilityType {
  RESIDENTIAL = 'RESIDENTIAL',
  COMMERCIAL = 'COMMERCIAL',
  MIXED_USE = 'MIXED_USE',
  INDUSTRIAL = 'INDUSTRIAL',
  HOSPITALITY = 'HOSPITALITY',
  EDUCATIONAL = 'EDUCATIONAL',
  GOVERNMENT = 'GOVERNMENT',
  OTHER = 'OTHER',
}
export interface Facility {
  id?: string;

  name: string;

  type: FacilityType;

  addressLine1: string;

  addressLine2?: string;

  city?: string;

  country?: string;

  postalCode?: string;

  description?: string;

  userRole: RoleName;
}

export interface CreateFacilityRequest {
  name: string;
  type: FacilityType;
  addressLine1: string;
  addressLine2?: string;
  city?: string;
  country?: string;
  postalCode?: string;
  description?: string;
  userRole: RoleName;
}

File: property-service/property-service/src/main/java/com/bnroll/property/entity/Facility.java
package com.bnroll.property.entity;

import com.bnroll.commercedomain.entity.BaseEntity;
import com.bnroll.dto.property.FacilityType;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "facility")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Facility extends BaseEntity {


    @Column(nullable = false, length = 150)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 30)
    private FacilityType type;

    @Column(nullable = false, length = 255)
    private String addressLine1;

    @Column(length = 255)
    private String addressLine2;

    @Column(length = 100)
    private String city;

    @Column(length = 100)
    private String country;

    @Column(length = 20)
    private String postalCode;

    @Column(length = 500)
    private String description;

    @Column(nullable = false)
    @Builder.Default
    private Boolean active = true;
}